AsyncDisplayKit/ASDisplayNode.mm

```diff
@@ -85,25 +87,94 @@ void ASDisplayNodeRespectThreadAffinityOfNode(ASDisplayNode *node, void (^block)
   }
 }
 
-+ (void)initialize
-{
-  if (self == [ASDisplayNode class]) {
-    return;
+/**
+ *  Returns ASDisplayNodeFlags for the givern class/instance. instance MAY BE NIL.
+ *
+ *  @param c        the class, required
+ *  @param instance the instance, which may be nil. (If so, the class is inspected instead)
+ *
+ *  @return ASDisplayNode flags.
+ */
+static struct ASDisplayNodeFlags GetASDisplayNodeFlags(Class c, ASDisplayNode *instance) {
+  struct ASDisplayNodeFlags flags = {0};
+
+  flags.isInHierarchy = NO;
+  flags.displaysAsynchronously = YES;
```

Comment:
Should we have assertions about the class being non-nil?  What if I passed in an ASScrollNode or something else that does not display asynchronously?  How about a developer creating something with initWithViewBlock/LayerBlock:, which also results in NO for displaysAsynchronously?